internal/dispatcher/nats.go

```diff
@@ -0,0 +1,40 @@
+package dispatcher
+
+import (
+	"log"
+
+	"github.com/nats-io/nats.go"
+)
+
+type NatsDispatcher struct {
+	connection *nats.Conn
+	connError  error
+}
+
+func (nd *NatsDispatcher) Publish(topic string, data []byte) error {
+	nc := nd.connection
```

Comment:
For simplicity of reading we should write something like that : 

```go
err := nd.connection.Publish(topic, data)
if err != nil {
log.Error("error publishing to topic", "error", err, "topic", topic)
return error
}
log.Debug("Published to topic", "topic", topic, "payload", data)
return nil
```

what do you think ?

Reply:
I think it makes sense, makes the code neater.